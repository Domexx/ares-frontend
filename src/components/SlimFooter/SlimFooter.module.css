/* === === === === SlimFooter === === === === */

.SlimFooter {
  width: 100%;
  margin-top: 50px;
  padding: 28px;
  text-align: center;
  background: var(--clr-dark-500);
  color: var(--clr-dark-200);
}

@media (--phone) {
  .SlimFooter {
    font-size: 12px;
  }
}
