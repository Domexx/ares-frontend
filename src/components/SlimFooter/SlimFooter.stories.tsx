import { Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';
import SlimFooterComponent, { Props } from './SlimFooter';

const Component = {
  title: 'Components/SlimFooter',
  component: SlimFooterComponent,
  argTypes: {},
};

const Template: Story<ComponentProps<typeof SlimFooterComponent>> = (args) => {
  return <SlimFooterComponent {...args} />;
};

export const SlimFooter = Template.bind({});
export default Component;

// https://storybook.js.org/docs/react/essentials/controls
SlimFooter.args = {
  copyright:
    'Â© 2022 Habbo.ac - is not affiliated with, endorsed, sponsored, or specifically approved by Sulake Oy or its Affiliates.',
} as Props;
