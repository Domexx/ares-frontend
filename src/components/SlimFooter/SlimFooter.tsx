import React from 'react';
import s from './SlimFooter.module.css';

export type Props = {
  copyright: string;
};

const SlimFooter: React.FC<Props> = ({ copyright }) => {
  return <footer className={s['SlimFooter']}>{copyright}</footer>;
};

export default SlimFooter;
