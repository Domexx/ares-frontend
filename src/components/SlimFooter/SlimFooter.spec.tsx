import React from 'react';

import { render } from '@testing-library/react';

import SlimFooter from './SlimFooter';

describe('Slim Footer Component', () => {
  it('should render properly', () => {
    const { container } = render(<SlimFooter copyright="copyright 2022" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
