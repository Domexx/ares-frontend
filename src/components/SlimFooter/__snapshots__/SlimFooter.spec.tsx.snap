// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slim Footer Component should render properly 1`] = `
<footer
  class="SlimFooter"
>
  copyright 2022
</footer>
`;
